transport/src/main/java/io/netty/channel/DefaultChannelPipeline.java

```diff
@@ -181,99 +200,103 @@ private void addLast0(AbstractChannelHandlerContext newCtx) {
         newCtx.next = tail;
         prev.next = newCtx;
         tail.prev = newCtx;
+        callHandlerAdded0(newCtx);
     }
 
     @Override
     public final ChannelPipeline addBefore(String baseName, String name, ChannelHandler handler) {
-        return addBefore(null, baseName, name, handler);
-    }
-
-    @Override
-    public final ChannelPipeline addBefore(
-            EventExecutor executor, String baseName, String name, ChannelHandler handler) {
-        final AbstractChannelHandlerContext newCtx;
         final AbstractChannelHandlerContext ctx;
-        synchronized (this) {
-            checkMultiplicity(handler);
-            name = filterName(name, handler);
-            ctx = getContextOrDie(baseName);
 
-            newCtx = newContext(executor, name, handler);
+        checkMultiplicity(handler);
+        if (name == null) {
+            name = generateName(handler);
+        }
 
-            addBefore0(ctx, newCtx);
+        final AbstractChannelHandlerContext newCtx = newContext(name, handler);
+        synchronized (handlers) {
+            int i = findCtxIdx((context) -> context.name().equals(baseName));
```

Comment:
single-arg lambda doesn't need parentheses, i.e. can be `just context -> ...` (same in other places)